plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -675,6 +675,22 @@ private static Map<String, String> getColumnNameToStatisticsName(JdbcTableHandle
         return columnNameToStatisticsName;
     }
 
+    @Override
+    protected String escapeObjectNameForMetadataQuery(String name, String escape)
+    {
+        requireNonNull(name, "name is null");
+        requireNonNull(escape, "escape is null");
+        checkArgument(!escape.isEmpty(), "Escape string must not be empty");
+        checkArgument(!escape.equals("_"), "Escape string must not be '_'");
+        checkArgument(!escape.equals("%"), "Escape string must not be '%'");
+        name = name.replace(escape, escape + escape);
+        name = name.replace("_", escape + "_");
+        name = name.replace("%", escape + "%");
+        name = name.replace("]", escape + "]");
+        name = name.replace("[", escape + "[");
```

Comment:
nit:

```suggestion
        // SQLServer also treats [ and ] as wildcard characters
        name = name.replace("]", escape + "]");
        name = name.replace("[", escape + "[");
```

Q: Do `[` and `]` still need escaping if they appear individually? e.g. `LIKE "a[column"` or `LIKE "a]column"`?

Reply:
tests added